File: libs_camera/src/main/java/com/xinruan/libs/camera/Camera2/MainActivity.java
package com.xinruan.libs.camera.Camera2;

import android.Manifest;
import android.app.Activity;
import android.content.Intent;
import android.content.pm.PackageManager;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.app.ActivityCompat;
import androidx.core.content.ContextCompat;

import com.xinruan.libs.camera.Camera2.GlsurfaceView.GlViewCameraActivity;
import com.xinruan.libs.camera.Camera2.SurfaceView.SurfaceViewCameraActivity;
import com.xinruan.libs.camera.Camera2.TexttureView.TexttureViewCameraActivity;
import com.xinruan.libs.camera.R;

/**
 * Created by GuoXu on 2020/9/9 16:16.
 */
public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        checkCameraPerssion();
        Button textureView = (Button) findViewById(R.id.textureView);
        Button surfaceView = (Button) findViewById(R.id.surfaceView);
        Button glview = (Button) findViewById(R.id.glview);
        textureView.setOnClickListener(this);
        surfaceView.setOnClickListener(this);
        glview.setOnClickListener(this);
    }

    /**
     * 权限检查
     *
     * @param mContext
     */
    public void checkCameraPerssion() {
        if (ContextCompat.checkSelfPermission(this, Manifest.permission.CAMERA) != PackageManager.PERMISSION_GRANTED) {
            requestPermission(this);
        } else {

        }
    }
    //检查和申请权限
    private void requestPermission(Activity mContext) {
        //把需要检查的一组权限放在一个字符串数组里
        String[] permissions = new String[]{Manifest.permission.CAMERA};
        //记录没有获得的权限数量
        //申请未获得的权限（直接把权限组传入）
        ActivityCompat.requestPermissions(mContext, permissions, 1);
    }


    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.textureView:
                Intent textureView = new Intent(this, TexttureViewCameraActivity.class);
                startActivity(textureView);
                break;
            case R.id.surfaceView:
                Intent surfaceView = new Intent(this, SurfaceViewCameraActivity.class);
                startActivity(surfaceView);
                break;
            case R.id.glview:
                Intent glview = new Intent(this, GlViewCameraActivity.class);
                startActivity(glview);
                break;
        }
    }
}

File: study_view/src/main/java/com/view/MainActivity.kt
package com.view

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import com.customRecyclerView.ListAdapter
import com.sn.study_pic.R
import com.view.recycleView.ItemListAdapter
import kotlinx.android.synthetic.main.activity_main.*

/**
 * Created by GuoXu on 2020/11/15 20:15.
 * 免findViewBuyID 插件 apply plugin: 'kotlin-android-extensions'
 */
class MainActivity : AppCompatActivity() {
    private lateinit var itemListAdapter: ItemListAdapter

    var itemName = arrayOf("拖拽View", "粘性View", "")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        initView();
    }

    private fun initView() {
        initRecyclerView()
    }

    private fun initRecyclerView() {
        recyclerview.layoutManager = LinearLayoutManager(baseContext)
        itemListAdapter = ItemListAdapter(baseContext)
        itemListAdapter.setData(itemName.toList())
        recyclerview.adapter = itemListAdapter
    }
}